### PR TITLE
[#D3D-5010] Adds the upload/export feature

### PR DESCRIPTION
**Pull request by loicSketchfab**

### CODE REVIEW

**Comment by loicSketchfab** [comment]
Switching to windows made me realize that this still needs a little bit of polishing...
Removing the "Needs review" label for now.

**Comment by loicSketchfab** [comment]
Fixes are now pushed and the addon works as expected on linux and windows.

**Comment by loicSketchfab** [comment]
Removing the "Needs review" label as the documentation is being worked on before the release.

File: addons/io_sketchfab_plugin/__init__.py [hunk 1]
```diff
@@ -570,13 +570,35 @@ def draw_search(layout, context):
...
-    ro = col.row();
+
+    ro = col.row()
+    ro.prop(props, "query")
+    ro.operator("wm.sketchfab_search", text="", icon='VIEWZOOM')
+
+
+    col = layout.box().column(align=True)
+    col.label(text="Filters")
+
```
Comment (AurL) [inline addons/io_sketchfab_plugin/__init__.py]: too many blank lines (check whole files)

File: addons/io_sketchfab_plugin/__init__.py [hunk 2]
```diff
@@ -1291,11 +1287,267 @@ def execute(self, context):
...
+            name="Title",
+            description="Title of the model (determined automatically if left empty)",
+            default="",
+            )
+
+# Singleton for storing global state
+class _SketchfabState:
+    __slots__ = (
+        "uploading",
+        "token_reload",
```
[the remark this answers is not in the document]
Comment (loicSketchfab) [reply addons/io_sketchfab_plugin/__init__.py]: Unused confirmed.

File: addons/io_sketchfab_plugin/__init__.py (hunk 1 above)
Comment (loicSketchfab) [reply addons/io_sketchfab_plugin/__init__.py]: Useless lines removed.

File: addons/io_sketchfab_plugin/__init__.py [hunk 3]
```diff
@@ -1291,11 +1287,267 @@ def execute(self, context):
...
+def upload_report(report_message, report_type):
+    sf_state.report_message = report_message
+    sf_state.report_type = report_type
+
+def _get_request_payload(data={}, files={}, json_payload=False):
+    """Helper method that returns the authentication token and proper content
+    type depending on whether or not we use JSON payload."""
+    headers = get_sketchfab_props().skfb_api.headers
+
+    if json_payload:
```
[the remark this answers is not in the document]
Comment (loicSketchfab) [reply addons/io_sketchfab_plugin/__init__.py]: Removed as indeed never used during the upload process.

**Pull request merged**